cmd/terramate/cli/cli.go

```diff
@@ -321,26 +424,47 @@ func (c *cli) initStack(dirs []string) error {
 
 func (c *cli) listStacks(mgr *terramate.Manager, isChanged bool) ([]terramate.Entry, error) {
 	if isChanged {
+		log.Trace().
+			Str("action", "listStacks()").
+			Str("stack", c.wd()).
+			Msg("`Changed` flag was set. List changed stacks.")
 		return mgr.ListChanged()
 	}
-
 	return mgr.List()
 }
 
 func (c *cli) printStacks() error {
+	logger := log.With().
+		Str("action", "printStacks()").
+		Logger()
+
+	logger.Trace().
+		Str("stack", c.wd()).
+		Msg("Create a new stack manager.")
 	mgr := terramate.NewManager(c.root(), c.prj.baseRef)
+
+	logger.Trace().
+		Str("stack", c.wd()).
+		Msg("Get stack list.")
 	entries, err := c.listStacks(mgr, c.parsedArgs.Changed)
 	if err != nil {
 		return err
 	}
 
+	logger.Trace().
+		Str("stack", c.wd()).
+		Msg("Print stacks.")
 	for _, entry := range entries {
 		stack := entry.Stack
 		stackRepr, ok := c.friendlyFmtDir(stack.Dir)
 		if !ok {
 			continue
 		}
 
+		logger.Debug().
+			Str("stack", c.wd()+stack.Dir).
+			Msg("Print stack.")
```

Comment:
```suggestion
		logger.Debug().
			Str("stack", stack.Dir).
			Msg("Print stack.")
```

stack.Dir is the canonical stack ID/path (relative to the project root). I think it is the more suitable one to log. If we want to log the full absolute path of a stack concatenating it with c.wd() is incorrect. Also when concatenating paths it is better to use [filepath.Join](https://pkg.go.dev/path/filepath#Join)